articles/fleetd-authentication.md

```diff
@@ -0,0 +1,103 @@
+- [Fleet desktop](https://fleetdm.com/guides/fleet-desktop#basic-article): This is the tray icon app to interface with users of the (Desktop) endpoint. This component can be disabled/missing on some endpoints (e.g. for servers without GUI).
+
+This document aims to explain how all three components authenticate to Fleet servers.
+
+## TLS
+
+All fleetd components communicate to Fleet using HTTPS.
+
+> Fleetd also supports mTLS, more on this below.
+
+## Enrollment to Fleet
+
+Teams in Fleet have an "enroll secret" that is used to authenticate endpoints during enrollment. 
+
+Enroll secrets are generated randomly by Fleet (if not provided) or can be defined by the admin.
+The enroll secrets ramdonly generated by Fleet are 24 bytes long, base64 encoded.
+
+### Fleetd package generation
+
+When generating a fleetd package using `fleetctl`, the admin provides an "enroll secret":
+```sh
+fleetctl package --type=pkg \
+    --fleet-desktop \
+    --fleet-url=... \
+    --enroll-secret=...
+```
+
+The "Fleet URL" and "Enroll secret" are embedded into the generated package.
+
+## Enrollment process in fleetd
+
+When installing a fleetd package the `orbit` binary is executed.
+`orbit` runs as `root` on macOS/Linux and as a `SYSTEM` service on Windows.
+
+During installation, the enroll secret is placed as a text file only accessible to `root`/`SYSTEM`.
+When `orbit` starts for the first time, it will attempt to enroll to Fleet using the enroll secret.
+If the enroll secret is valid, then Fleet will generate a random "orbit node key" and return it to `orbit`.
+`orbit` will store such "node key" on a text file only accessible to `root`/`SYSTEM`. After enrollment, the "orbit node key" will be used to authenticate orbit related requests to Fleet.
+
+`orbit` will start the `osquery` daemon (that also runs as `root`/`SYSTEM`).
+The `osquery` daemon will enroll to Fleet using the same enroll secret used by `orbit`.
+Similarly, if the enroll secret is valid, then Fleet will generate a random "osquery node key" and return it to `osquery`.
+`osquery` will store such "node key" on its internal RocksDB database file, only accessible to `root`/`SYSTEM`.
+After enrollment, the "osquery node key" will be used to authenticate osquery related requests to Fleet.
+
+On macOS and Windows, once the `orbit` and `osquery` enrollment process is completes, the enroll secret is removed from the file system and stored in the Apple keychain and Windows credentials manager respectively.
+
+After starting osquery, `orbit` will also start the `Fleet desktop` tray application. Such application runs as the Desktop GUI user (it does not run as `root`/`SYSTEM`). Before starting `Fleet desktop`, `orbit` will set a random "Fleet desktop token" and send it to Fleetthe `orbit` authenticated communication channel. `orbit` stores such token on a file accessible to the GUI user and `Fleet desktop` will use such token to authenticate requests to Fleet.
```

Comment:
@lucasmrod I think there is a typo here: send it to Fleetthe `orbit` authenticated

Reply:
Thanks! Should be fixed now.